collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderActivity.java

```diff
@@ -116,6 +155,12 @@ protected void onCreate(Bundle savedInstanceState) {
             showDialog(PROGRESS_DIALOG);
             instanceServerUploader = new InstanceServerUploader();
 
+            // The 3 details - custom URL, username & password passed through the intent should be available if we are to override the default settings
+            // If one of them lacks (in this case it's a null) then we do not override the form or device settings during submission
+            if (url != null && username != null && password != null) {
```

Comment:
Oh, and I guess everywhere, really. It looks like this was a very deliberate choice so would be good to understand your reasoning.

Reply:
Yea, it was a deliberate choice before I realized that anonymous logins are possible. I also assumed empty strings for the username and password would do ok for anonymous logins as per this 

>  Does passing in the URL and empty strings for username and password work?

 I'll work on changing that 😄